src/groovy/org/pih/warehouse/importer/CSVUtils.groovy

```diff
@@ -63,11 +63,49 @@ class CSVUtils {
     static int parseInteger(String s, String fieldName = "unknown_field") {
         try {
             return parseNumber(s, fieldName).intValueExact()
-        } catch (ArithmeticException e) {
+        } catch (ArithmeticException ignored) {
             throw new IllegalArgumentException("Expected integer value for ${fieldName}=${s}")
         }
     }
 
+    /**
+     * Format an identifier for inclusion in a CSV file.
+     *
+     * @param id an arbitrary identifier encoded as a string -- ok if null
+     * @return a string representation of `id` suitable for CSV export
+     *
+     * You might wonder why converting a string to a string would be
+     * needed for CSV export. Well, it turns out that Excel's CSV
+     * importer will parse, say, the ZIP code 02199, as the number 2199.
+     */
+    static String formatId(String id) {
+
+        if ((id?.startsWith("0") && id?.isNumber()) ||
+            (id ==~ /^[\d.-\/]+$/) && (id?.contains("-") || id?.contains("/"))) {
+            /*
+             * If the supplied ID could be interpreted as a number or as
+             * a numerical date, why not just prefix it with a literal tab?
+             * (Apparently, this tells Excel not to attempt to parse it.)
+             *
+             * https://superuser.com/questions/318420/formatting-a-comma-delimited-csv-to-force-excel-to-interpret-value-as-a-string
+             *
+             * Unfortunately, no other spreadsheet software does this :-(
+             *
+             * The following kludge plays nicely enough with Excel, Google
+             * Sheets, LibreOffice, and Numbers (Mac), but it could easily
+             * confuse other CSV parsers. (Not sure what we can do: Excel
+             * is our most important client software, by far.)
+             */
+            return "=\"${id}\"".toString()
```

Comment:
I don't mind this, but I believe this is handled by the StringEscapeUtils 

https://commons.apache.org/proper/commons-text/javadocs/api-release/org/apache/commons/text/StringEscapeUtils.html#escapeCsv(java.lang.String)

Reply:
@jmiranda I'm afraid it's worse than it appears.

This returned string actually is intended as _input_ to the StringEscapeUtils, which the caller takes care of.

The ultimate contents of a csv record for a field that starts with a 0 ends up being `"=""00912"""`. If you just call escapeCSV, Excel will ignore your attempted escape, treat the ID as a number, and trim the leading 0's. :-(